models/content.go

```diff
@@ -0,0 +1,76 @@
+// and is used for marshaling and unmarshaling json representation for API
+// ID is a duplicate of ID in TopicResponse, to facilitate each subdocument being a full-formed
+// response in its own right depending upon request being in publish or web and also authentication.
+type Content struct {
+	State               string            `bson:"state,omitempty"                 json:"state,omitempty"`
+	Spotlight           *[]TypeLinkObject `bson:"spotlight,omitempty"             json:"spotlight,omitempty"`
+	Articles            *[]TypeLinkObject `bson:"articles,omitempty"              json:"articles,omitempty"`
+	Bulletins           *[]TypeLinkObject `bson:"bulletins,omitempty"             json:"bulletins,omitempty"`
+	Methodologies       *[]TypeLinkObject `bson:"methodologies,omitempty"         json:"methodologies,omitempty"`
+	MethodologyArticles *[]TypeLinkObject `bson:"methodology_articles,omitempty"  json:"methodology_articles,omitempty"`
+	StaticDatasets      *[]TypeLinkObject `bson:"static_datasets,omitempty"       json:"static_datasets,omitempty"`
+	Timeseries          *[]TypeLinkObject `bson:"timeseries,omitempty"            json:"timeseries,omitempty"`
+}
+
+// TypeLinkObject represents a generic structure for all type links
+type TypeLinkObject struct {
+	HRef  string `bson:"href,omitempty"   json:"href,omitempty"`
+	Title string `bson:"title,omitempty"  json:"title,omitempty"`
+}
+
+// PrivateContentItem represents an evolving content with the current content and the updated content.
+// This is for the REST API response.
+// The 'Next' is what gets updated throughout the publishing journey, and then the 'publish' step copies
+// the 'Next' over the 'Current' document, so that 'Current' is whats always returned in the web view.
+type PrivateContentItem struct {
+	Next    *ContentItem `json:"next,omitempty"`
+	Current *ContentItem `json:"current,omitempty"`
+}
+
+// PublicContent used for returning just the Current document(s) in REST API response
+type PublicContent struct {
+	Count       int            `json:"count"`
+	Offset      int            `json:"offset_index"`
+	Limit       int            `json:"limit"`
+	TotalCount  int            `json:"total_count"`
+	PublicItems []*ContentItem `json:"items"`
+}
+
+// PrivateContent used for returning both Next and Current document(s) in REST API response
+type PrivateContent struct {
+	Count        int                   `json:"count"`
+	Offset       int                   `json:"offset_index"`
+	Limit        int                   `json:"limit"`
+	TotalCount   int                   `json:"total_count"`
+	PrivateItems []*PrivateContentItem `json:"items"`
+}
+
+// ContentItem is an individual content item
+type ContentItem struct {
+	Title string        `json:"title,omitempty"`
+	Type  string        `json:"type,omitempty"`
+	Links *ContentLinks `json:"links,omitempty"`
+	State *bool         `json:"state,omitempty"` //!!! Eleanor ... do we need something like this ?
```

Comment:
is this something that @eldeal  should check? Or can we approve it for now and change it later if required?

Reply:
This can be aprovved for now and its 'up for adjustment' later when Eleanor is back.